intake_esm/core.py

```diff
@@ -225,6 +229,13 @@ def __getitem__(self, key):
         else:
             raise KeyError(key)
 
+    def _get_entries(self):
+        """Eager evaluation of the delayed entries dictionary.
+           This is needed for enabling functionality provided by the base class
+           which expects a regular dictionary."""
+        entries = dask.compute(self._entries)[0]
+        return entries
```

Comment:
I was thinking of something a bit more lightweight (I don't think we want to involve dask at this point).

I'm not 100% sure how to do this but I'm thinking a just-in-time entry creation would work:

```python
def __getitem__(self, key):
    try:
        return self._entries[key]
    except KeyError:
        if key in self.keys():
            self._entries[key] = make_entry(key)
            return self._entries[key]
        else:
            raise KeyError
```
Do you think something like this could be made to work?

Reply:
> Do you think something like this could be made to work?

I think so. We may need to override a few methods from the base class. I will look into it, and will get back to you.

Reply:
@jhamman, should `__getitem(self, key)__` return an Entry instead of an `esm_datastore()` object?

Reply:
Per [intake](https://github.com/intake/intake/blob/cd3873e2933327e87e095e0ea653aacca72ab126/intake/catalog/base.py#L142), 

        * ``__iter__()`` -> an iterator of entry names
        * ``__getitem__(key)`` -> an Entry
        * ``items()`` -> an iterator of ``(key, Entry)`` pairs
        For best performance the object should also implement:
        * ``__len__()`` -> int
        * ``__contains__(key)`` -> boolean

Reply:
@jhamman, the latest commit address the just in time entry creation.